client/checkout/woopay/express-button/woopay-express-checkout-button.js

```diff
@@ -84,14 +82,77 @@ export const WoopayExpressCheckoutButton = ( {
 		}
 	}, [ isPreview, context ] );
 
-	const defaultOnClick = useCallback( ( event ) => {
-		// This will only be called if user clicks the button too quickly.
-		// It saves the event for later use.
-		initialOnClickEventRef.current = event;
-		// Set isLoadingRef to true to prevent multiple clicks.
-		isLoadingRef.current = true;
-		setIsLoading( true );
-	}, [] );
+	const canAddProductToCart = useCallback( () => {
+		if ( ! isProductPage ) {
+			return true;
+		}
+
+		const addToCartButton = document.querySelector(
+			'.single_add_to_cart_button'
+		);
+
+		if (
+			addToCartButton &&
+			( addToCartButton.disabled ||
+				addToCartButton.classList.contains( 'disabled' ) )
+		) {
+			if (
+				addToCartButton.classList.contains(
+					'wc-variation-is-unavailable'
+				)
+			) {
+				window.alert(
+					window?.wc_add_to_cart_variation_params
+						?.i18n_unavailable_text ||
+						__(
+							'Sorry, this product is unavailable. Please choose a different combination.',
+							'woocommerce-payments'
+						)
+				);
+			} else if (
+				addToCartButton.classList.contains(
+					'wc-variation-selection-needed'
+				)
+			) {
+				window.alert(
+					window?.wc_add_to_cart_variation_params
+						?.i18n_make_a_selection_text ||
+						__(
+							'Please select some product options before adding this product to your cart.',
```

Comment:
Is this message approved by the design team? Since it's guiding users to the payment screen rather than adding items to the cart, clarity is key. Perhaps a concise alternative like `Configure your product options to proceed to payment.` would more accurately reflect the user action. What do you think?

Also, I find the use of window alerts a bit unconventional, but I see it's consistent with other PRBs. So, let's stick with it for now for consistency's sake 🤷🏻‍♂️.

Reply:
> Is this message approved by the design team?

Actually, this isn't out of nowhere. It's the same text from `window.wc_add_to_cart_variation_params.i18n_make_a_selection_text`, but a fallback in case the variable is not defined for some reason. Also, the other PRB use the same text.

The behavior should be the same before this PR which fixes a regression, but I agree with the point that "adding to cart" is not exactly the right action 👍. I will ask in the issue.

> Also, I find the use of window alerts a bit unconventional

I feel the same, but it's more of a Woo Core [default behavior](https://github.com/woocommerce/woocommerce/blob/8.0.1/plugins/woocommerce/client/legacy/js/frontend/add-to-cart-variation.js#L130) and design issue which is more complex to change than it looks (see https://github.com/Automattic/woocommerce-payments/issues/6634#issuecomment-1680322978).